toolchain/check/impl_lookup.cpp

```diff
@@ -130,6 +131,38 @@ auto LookupImplWitness(Context& context, SemIR::LocId loc_id,
     }
   }
 
+  auto& stack = context.impl_lookup_stack();
+  // Deduction of the interface parameters can do further impl lookups, and we
+  // need to ensure we terminate.
+  //
+  // https://docs.carbon-lang.dev/docs/design/generics/details.html#acyclic-rule
+  // - We look for violations of the acyclic rule by seeing if a previous lookup
+  //   had all the same type inputs.
+  // - The `interface_const_id` encodes the entire facet type being looked up,
+  //   including any specific parameters for a generic interface.
+  //
+  // TODO: Implement the termination rule, which requires looking at the
+  // complexity of the types on the top of (or throughout?) the stack:
+  // https://docs.carbon-lang.dev/docs/design/generics/details.html#termination-rule
+  for (auto entry : stack) {
+    if (entry.type_const_id == type_const_id &&
+        entry.interface_const_id == interface_const_id) {
+      CARBON_DIAGNOSTIC(ImplLookupCycle, Error,
+                        "cycle found in lookup of interface {0} for type {1}",
+                        std::string, SemIR::TypeId);
+      context.emitter()
+          .Build(loc_id, ImplLookupCycle, "<TODO: interface name>",
+                 context.GetTypeIdForTypeConstant(type_const_id))
+          .Emit();
+      return SemIR::ErrorInst::SingletonInstId;
+    }
+  }
+  stack.push_back(Context::ImplLookupStackEntry{
```

Comment:
```suggestion
  stack.push_back({
```

Style: https://github.com/carbon-language/carbon-lang/blob/trunk/docs/project/cpp_style_guide.md#syntax-and-formatting

"Prefer to only include the typename when required to compile (WizType{.a = 1})."

Reply:
Ah yes thanks